src/player.cpp

```diff
@@ -11168,12 +11177,26 @@ activate your weapon."), gun->tname().c_str(), _(mod->location.c_str()));
 
         if (choice < int(mods.size())) {
             const std::string mod = used->contents[choice].tname();
-            remove_gunmod(used, unsigned(choice));
-            add_msg(_("You remove your %s from your %s."), mod.c_str(), used->tname().c_str());
+
+            if( used->contents[choice].has_flag("IRREMOVABLE") ){
+                add_msg(_("You cannot remove integrated mods.") );
+            }
+            else{
+                remove_gunmod(used, unsigned(choice));
```

Comment:
no need to `unsigned()` this afaik, its usage above as a `size_t` is evident, and any check for improper choice should catch it. if it's bad, you'll get a bad index warning regardless if it's unsigned or not, the result is still oob.